docs/docs/noir/standard_library/mem.md

```diff
@@ -0,0 +1,52 @@
+---
+title: Memory Module
+description:
+  This module contains functions which manipulate memory in a low-level way
+keywords:
+  [
+    mem, memory, zeroed, transmute, checked_transmute
+  ]
+---
+
+# `std::mem::zeroed`
+
+```rust
+fn zeroed<T>() -> T
+```
+
+Returns a zeroed value of any type.
+This function is generally unsafe to use as the zeroed bit pattern is not guaranteed to be valid for all types.
+It can however, be useful in cases when the value is guaranteed not to be used such as in a BoundedVec library implementing a growable vector, up to a certain length, backed by an array.
+The array can be initialized with zeroed values which are guaranteed to be inaccessible until the vector is pushed to.
+Similarly, enumerations in noir can be implemented using this method by providing zeroed values for the unused variants.
+
+This function currently supports the following types:
+
+- Field
+- Bool
+- Uint
+- Array
+- Slice
+- String
+- Tuple
+- Functions
+  
+Using it on other types could result in unexpected behavior.
+
+# `std::mem::checked_transmute`
+
+```rust
+fn checked_transmute<T, U>(value: T) -> U
+```
+
+Transmutes a value of one type into the same value but with a new type `U`.
+
+This function is safe to use since both types are asserted to be equal later during compilation.
```

Comment:
`later during compilation` -> `when all type-level variables are known`?

Reply:
Changed it to "... after the concrete values for generic types become known"